frontend/multivac_site.py

```diff
@@ -1,14 +1,126 @@
-from flask import Flask, render_template
+import logging
+
+from flask import Flask, make_response, render_template, request
 
 import frontend.frontend_constants as constants
 
+from agents.agent_registry import AGENTS
+from environment.environment_registry import ENVIRONMENTS
+from session.session_starter import start_multivac_session
+from session.session_status_enum import SessionStatusEnum
+
 
 app = Flask(__name__)
+app.config['TEMPLATES_AUTO_RELOAD'] = True
+
+logger = logging.getLogger(__name__)
+logger.addHandler(logging.StreamHandler())
+logger.setLevel(logging.DEBUG)
+
+
+@app.route('/', methods=['GET', 'POST'])
+def route_index_page():
+    """
+    This is the main routing function for the Multivac page.
+
+    For a POST request, assume the launcher form has been submitted and render the session page.
+    For a GET request or any other request, render the launcher form.
+    """
+
+    if request.method == 'POST':
+        # Read in post parameters. If key is not located in the request, this will raise an exception.
```

Comment:
Validate request.form contains these parameters and can be appropriately cast. Otherwise, return a 400 error